Paper: The Geoscience Community Analysis Toolkit: An Open Development, Community Driven Toolkit in the Scientific Python Ecosystem

erogluorhan opened the pull request:
Draft for now, aimed at generating the PDF only.

If you are creating this PR in order to submit a draft of your paper,
see http://procbuild.scipy.org/ for logs generated by the build
process.

See the [project readme](https://github.com/scipy-conference/scipy_proceedings#build-your-paper)
for more information.


*Editor*: <!--editor-->@cbcunc<!--end-editor-->

*Reviewers*: <!--reviewers-list-->@RBhupi, @TomNicholas<!--end-reviewers-list-->

erogluorhan committed:
Initial look of the paper

scoobies commented:
Hi!, I'm @scoobies, a friendly bot.

Type ```@scoobies help``` to discover how I can help you.

erogluorhan committed:
Empty to trigger PDF generator

erogluorhan committed:
Try to fix PDF generation error

erogluorhan committed:
Fix one citation

erogluorhan committed:
Added more citations

erogluorhan committed:
Fix references

erogluorhan committed:
Fix references

erogluorhan committed:
Fix references

erogluorhan committed:
Fix references

erogluorhan committed:
Fix references

erogluorhan committed:
Try capitalizing in bib file

erogluorhan committed:
Some more corrections in bib

erogluorhan committed:
Added text and figure into software title

erogluorhan committed:
Fix the figure

erogluorhan committed:
Text and figure updates

erogluorhan committed:
Fix figure labels

erogluorhan committed:
Update the text

erogluorhan committed:
Added a new figure repo structure

erogluorhan committed:
GeoCAT-comp part finished

erogluorhan committed:
Added GeoCAT-examples and -viz and a figure

erogluorhan committed:
Fix fig4 width

erogluorhan committed:
A few typo kind of corrections

erogluorhan committed:
Added a GeoCAT-examples figure and WRF-python text

erogluorhan committed:
Initial text in Project Raijin

erogluorhan committed:
More Raijin text

erogluorhan committed:
More text and a figure into Raijin

erogluorhan committed:
Raijin text and figure

erogluorhan committed:
A few rephrasing

erogluorhan committed:
Open dev and pythia text

erogluorhan committed:
Address reviews about Introduction

erogluorhan committed:
Incorporate @michaelavs's reviews

erogluorhan committed:
Incorporate Anissa's reviews

erogluorhan committed:
Fix references

erogluorhan committed:
Fix bibs

erogluorhan committed:
Fix bibs

erogluorhan committed:
Try fiixing bibs

erogluorhan committed:
Try fixing bibs

erogluorhan committed:
Try fixing bibs

erogluorhan committed:
Try fixing bibs

erogluorhan committed:
Try fixing bibs

erogluorhan committed:
Try fixing bibs

erogluorhan committed:
Try fixing bibs

erogluorhan committed:
Try fixing bibs

erogluorhan committed:
Try fixing bibs

erogluorhan committed:
Try fixing bibs

erogluorhan committed:
Address reordering request for 2nd paragraoh of GeoCAT-comp

erogluorhan committed:
Cont'd addressing of reordering request for 2nd paragraoh of GeoCAT-comp

erogluorhan committed:
Cont'd addressing of reordering request for 2nd paragraoh of GeoCAT-comp

erogluorhan committed:
Cont'd addressing of reordering request for 2nd paragraoh of GeoCAT-comp

erogluorhan committed:
Add Numba mentioning as suggested

erogluorhan committed:
Finalize Numba mention

erogluorhan committed:
Address a review request about badges paragraph

erogluorhan committed:
Update how to show all the badges up

erogluorhan committed:
Scalability title and text added

erogluorhan committed:
Reorder authors

erogluorhan committed:
Correct a typo

deniederhut commented:
@scoobies mark paper

deniederhut commented:
@scoobies build status

scoobies commented:
{"erogluorhan-2022"=>{"data"=>{"build_output"=>"[*] Updating proceedings build tools in {self.master_repo_path}...\nAlready up to date.\n[*] Check out paper repository...\nCloning into '/tmp/tmpa02xgi3u'...\nMoving proceedings build tools to temp directory...\n[*] Build the paper...\nBuilding: orhan_eroglu\n*** Warning: /tmp/tmpa02xgi3u/publisher/../output/orhan_eroglu/paper_stats.json does not exist.\n", "build_pdf_path"=>"/procbuild/procbuild/../cache/erogluorhan-2022.pdf", "build_status"=>"success", "build_timestamp"=>"28/05 15:46"}, "status"=>"success"}}

deniederhut commented:
@scoobies mark needs review

deniederhut commented:
@scoobies assign @cbcunc as editor

scoobies commented:
Assigned! @cbcunc is now the editor

mepa commented:
@scoobies add @RBhupi to reviewers

scoobies commented:
@RBhupi added to the reviewers list!

cbcunc commented:
Editor confirms paper builds.

deniederhut commented:
Hi! We've moved the deadline for first reviews back one week, to June 16th, to give us a little more time to find reviewers for all the papers this year. You can see the updated full timeline at https://github.com/scipy-conference/scipy_proceedings#timeline-for-2022.

deniederhut commented:
@scoobies add @TomNicholas to reviewers

scoobies commented:
@TomNicholas added to the reviewers list!

RBhupi commented on papers/orhan_eroglu/00_geocat.rst:
```diff
@@ -0,0 +1,446 @@
+but has not been actively developed as it has been in maintenance mode.
+
+The initiative had an initial two-year roadmap with major milestones being: (1)
+Replicating NCL’s computational routines in Python, (2) training and support for
```
The geoCAT blog and documentation suggest that a lot of development has been made in achieving the first goal of geoCAT. In the following section (GeoCAT Software), please consider providing more quantitative details of the development since the ‘Pivot to python’. For example, 'the WRF functions, interpolation, and plotting functions are 100% completed while functions in the categories of thermodynamics are x of y finished'.  A quantitative measure of the progress of GeoCAT should give a good idea of what has been achieved and what remains. This will also help identify and discuss the specific topics where GeoCAT needs support from the python community.

erogluorhan replied:
Thanks very much for this suggestion! Please see this [commit](https://github.com/scipy-conference/scipy_proceedings/pull/686/commits/768a86a4988219d9ebf5afcbcf0b1f0b19ab3db4) in which we attempted to address the recommended additions.

deniederhut commented:
@scoobies mark needs more review

RBhupi commented on papers/orhan_eroglu/00_geocat.rst:
```diff
@@ -0,0 +1,446 @@
+on contributors of pure Python code to build, compile, or test any compiled
+code (e.g. Fortran) at GeoCAT-comp level.
+
+GeoCAT-examples (and GeoCAT-viz)
```
Why is geoCAT-example considered as a tool while it is actually the documentation of geoCAT tools?

erogluorhan replied:
GeoCAT-examples is not actually the documentation of the GeoCAT tools. Instead, it is the visualization component of the GeoCAT umbrella, and it not only houses more than 140 plotting scripts and several GeoCAT-comp example visualizations, but also provides high level wrapper functions on top of matplotlib and cartopy with the help of GeoCAT-viz. We thought we put this decription clearly int the paper, but please let us know if you think some specific statements from the manuscript leads the reader to think GeoCAT-examples is for GeoCAT documentation.

RBhupi commented on papers/orhan_eroglu/00_geocat.rst:
```diff
@@ -0,0 +1,446 @@
+   Taylor diagram and curly vector examples that created with the help
+   of GeoCAT-viz :label:`fig5viz`
+
+WRF-Python
```
Why is WRF-Python not part of Fig 3?

erogluorhan replied:
Now updated the Figure (it's now Figure 4) and added WRF-Python as well as Project Raijin.

RBhupi commented on papers/orhan_eroglu/00_geocat.rst:
```diff
@@ -0,0 +1,446 @@
+   :cite:`geocatcontribguide` :label:`fig3structure`
+
+Many of the computational functions in GeoCAT are implemented in pure Python.
+However, there are others that are implemented in Fortran but wrapped up
```
NumPy is superfast for most computations. Any reason to choose FORTRAN for some task?

TomNicholas replied:
Presumably wrapping existing fortran functions so that they could be called from python (using [f2py](https://numpy.org/doc/stable/f2py/index.html)) was easier than re-implementing those functions in python.

erogluorhan replied:
Thanks, @TomNicholas, that was exactly how we initially approached some of the implementation cases. There was also a second motivation we added now to the paper. @RBhupi , to make this clarified, we added some details in this [commit](https://github.com/scipy-conference/scipy_proceedings/pull/686/commits/58f440bbc5dbd5d53542bccc461deb6f8f3ba2ab). Please let us know if that suffice or not.

RBhupi commented on papers/orhan_eroglu/00_geocat.rst:
```diff
@@ -0,0 +1,446 @@
+
+- Fourier transforms such as band-block, band-pass, low-pass, and high-pass
+
+- Meteorological variables such as relative humidity, dew-point temperature,
```
'Meteorological variables' Did you mean 'Meteorological functions' to compute the listed variables?

erogluorhan replied:
Thanks for catching this! Now corrected to "Meteorological variable computations"

RBhupi commented on papers/orhan_eroglu/00_geocat.rst:
```diff
@@ -0,0 +1,446 @@
+changes such as font sizes, color schemes, etc. that resemble NCL plots),
+the GeoCAT-viz library :cite:`geocatvizrepo` was implemented. Use of
+functions from this library in GeoCAT-examples significantly reduces the
+LOC requirements for most of the visualization examples to comparable
```
As stated above the Matplotlib and Cartopy (M&C) takes 100s of LOC while NCL and GeoCAT only take 10s. Providing LOC count for M&C, NCL, and Geocat for the shown examples will strengthen this argument.

erogluorhan replied:
Thanks for asking this! We have now added text to make a comparison. 

FYI: Taylor diagram scripts in GeoCAT-examples, for instance, were never implemented by using directly the M&C functions (i.e. they were implemented using our GeoCAT-viz functions). Almost all of the others are the same (GeoCAT-viz and GeoCAT-examples efforts were simultaneously born as the result of a multiple day hackathon in 2020; thus, we used GeoCAT-viz functions from the day one even though we made some improvements to it in the meantime). Therefore, giving an exact number for replicated NCL examples using direct M&C codes is not easy, but we did our best to give a good example in the manuscript to show where it would end up if we did call M&C functions directly. 

Please let us know if you think further examples/clarification would be needed in the manuscript.

RBhupi commented on papers/orhan_eroglu/00_geocat.rst:
```diff
@@ -0,0 +1,446 @@
+community involvement at any level of the project lifecycle alongside having the
+whole software stack open-sourced.
+
+GeoCAT has created several, now-established, Python tools that are hosted and
```
This sentence can be broken into two or be shortened. Consider "GeoCAT has six(or seven) Python tools for geoscientific computation and visualization which are built on Pangeo packages such as Xarray, and Dass, as well as Jupyter Notebooks."

erogluorhan replied:
Thanks for the suggestion, we made changes as recommended.

RBhupi commented on papers/orhan_eroglu/00_geocat.rst:
```diff
@@ -0,0 +1,446 @@
+(HPC) systems such as NCAR’s Casper, Cheyenne, and upcoming Derecho clusters
+:cite:`ams22craker`. Additionally, GeoCAT also utilizes Numba, an open source
+just-in-time (JIT) compiler :cite:`numba`, to translate Python and NumPy code into
+machine codes in order to get faster executions wherever possible.
```
The above paragraph is trying to explain too many technical aspects of several tools and dependencies, but it is difficult to comprehend by the non-developers. Therefore, I may have misunderstood this completely. Consider adding a diagram to make this paragraph more readable.

erogluorhan replied:
We have added Fig 1 (the newer first figure in the paper) in an attempt to ease the understanding of the roles of those technologies. We hope it helps a bit.

RBhupi commented on papers/orhan_eroglu/00_geocat.rst:
```diff
@@ -0,0 +1,446 @@
+   :scale: 50%
+   :figclass: w
+
+   GeoCAT project structure with all of the software tools
```
The tool GeoCAT-datafiles is not explained in the paper.

erogluorhan replied:
Added now, please see the commit [c02e915](https://github.com/scipy-conference/scipy_proceedings/pull/686/commits/c02e9151cc2760792622996aa8a95874470ae87f) (We originally thought it was a lightweight data storage and access tool, and we shouldn't add it to the manuscript, but we agree it is best to mention it shortly for the sake of comprehension).

RBhupi reviewed (changes requested):
The paper discusses the development of the python GeoCAT toolkit established to provide a python alternative to NCL users in the geoscience community. The authors provided an overview of their CI/CD practices, documentation, and source code links, as well as a description of the geoCAT components, and community engagement efforts. The geoCAT blog posts and documentation suggest that a lot of development has been made in achieving the first goal of GeoCAT. The paper also shows NCL-like plotting ability with examples. 
Other developers in the SciPy community, will value the specific examples of learning experiences of the GeoCAT team while porting the NCL to python. Overall the paper is well written however, the sections “Introduction” and  “GeoCAT Software” will benefit from more specific and concrete words and details in the description of the tools. Please check the specific comments.

RBhupi commented on papers/orhan_eroglu/00_geocat.rst:
```diff
@@ -0,0 +1,446 @@
+- Empirical orthogonal function (EOF) analysis
+
+
+.. figure:: figures/fig3_structure.png
```
I think this flowchart is complicated due to the groupings of the tools that don't group together nicely. I assume, GeoCAT-datafiles is IO, not visualization, and GeoCAT-examples are applications of the python tools, they are not tools themselves. Most example scripts have NCL-like Read-Compute-Visualize structure. See if the attached call graph makes sense 
![GeoCAT1](https://user-images.githubusercontent.com/2460536/173217819-48d0e0aa-df2b-4286-8dcc-c7c124903366.png)

erogluorhan replied:
You're right; thanks for catching this! We updated the flowchart (also to address the lack of WRF-Python and UXarray from the flowchart)

deniederhut commented:
Hey @TomNicholas ! Friendly reminder that reviews are due tomorrow

TomNicholas commented on papers/orhan_eroglu/00_geocat.rst:
```diff
@@ -0,0 +1,446 @@
+platforms (e.g. AWS).
+
+In order to keep up with the scalability objectives, GeoCAT functions are
+implemented to operate on Dask DataArray objects in addition to natively
+supporting Xarray and NumPy data arrays. Therefore, the GeoCAT functions can
```
**Minor nitpick**: This is not quite accurate - technically there is no such object as a "Dask DataArray". 

Rather, `xarray.DataArray` objects can wrap either `dask.Array` objects or `numpy.ndarray` objects. A "chunked xarray DataArray" is an example of the former. The portmanteau "DataArray" is only used to describe xarray objects, and means something different from "data array", which could be referring to any number of types.

This distinction can be important as other parts of the ecosystem work a little differently - there is for example a "Dask Dataframe". It also matters when one wants to start composing with other array types - as for example we do with [`pint-xarray`](https://github.com/xarray-contrib/pint-xarray).

It would be more correct to speak of "dask-backed xarray DataArrays", "dask arrays", or "numpy arrays".

erogluorhan replied:
Thanks very much for catching this and providing a thorough clarification! We addressed this suggestion; please let us know if you suggest further terminology clarification here.

TomNicholas commented on papers/orhan_eroglu/00_geocat.rst:
```diff
@@ -0,0 +1,446 @@
+----------------
+
+To ensure community involvement at every level in the development lifecycle,
+GeoCAT is committed to an open development model. With this model, we not only
+have our project stack open-sourced but also ensure most of the project assets
+that are directly related to the software development lifecycle are publicly
```
It would be nice to define the term "open development" earlier in the manuscript, because I was distracted wondering exactly what it meant (and if it was different from open-source) until I got here.

erogluorhan replied:
Thanks! We now moved the definition to the abstract, and we believe both parts (abstract and open development section) now can provide better understanding.

TomNicholas commented on papers/orhan_eroglu/00_geocat.rst:
```diff
@@ -0,0 +1,446 @@
+meetings are great venues for such a scientific software engineering project
+to share updates and progress with the community. For instance, the American
+Meteorological Society (AMS) Annual Meeting and American Geophysical Union
+(AGU) Fall Meeting are two significant scientific events that GeoCAT has
+always been presented by one or multiple publications since its birth to
+inform the community. The annual Scientific Computing with Python (SciPy)
```
Grammar - should be "the GeoCAT team has presented at" or similar.

erogluorhan replied:
Fixed now, thanks!

TomNicholas commented on papers/orhan_eroglu/00_geocat.rst:
```diff
@@ -0,0 +1,446 @@
+aligning deployments with the project milestones as well as SPE
+requirements and specifications.
+
+GeoCAT has created the Xarray-based Uxarray package :cite:`uxarrayrepo` to
```
Do you want to add uxarray to [xarray's ecosystem list?!](https://github.com/pydata/xarray/blob/main/doc/ecosystem.rst)

erogluorhan replied:
That would be great! Even GeoCAT-comp could be a good candidate to be listed there.

TomNicholas commented on papers/orhan_eroglu/00_geocat.rst:
```diff
@@ -0,0 +1,446 @@
+
+GeoCAT has created several, now-established, Python tools that are hosted and
+managed publicly on GitHub to develop computation and visualization functions,
+which are built on cornerstone Pangeo :cite:`pangeo18` (i.e. a community platform
+for big data geoscience) packages such as Xarray :cite:`xarray17`, Dask
```
I would separate out the sentence about pangeo to make this text clearer. 

"GeoCAT builds upon (or arguably is a member of) the Pangeo ecosystem. In particular it relies upon xarray, ..."

erogluorhan replied:
Tried to address this, also combining with @RBhupi 's previous recommendation. Please let us know if the current structure of sentences does not work.

TomNicholas commented on papers/orhan_eroglu/00_geocat.rst:
```diff
@@ -0,0 +1,446 @@
+Briefly, GeoCAT-comp houses computational operators for applications ranging from
+regridding and interpolation, to climatology and meteorology. GeoCAT-examples
+provides over 140 publication-quality plotting scripts in Python for Earth
+sciences. It also houses Jupyter notebooks with high-performance, interactive
+plots that enable features such as pan and zoom on fine-resolution geoscience
+data (e.g. ~3 km data rendered within a few tens of seconds to a few minutes on
```
I would like to know more about how you do interactive plotting of high-resolution datasets. How does this work? Are the values to be displayed lazily re-computed with dask.delayed objects? Can matplotlib actually handle large images?

This is a space in which there are lots of exciting new tools (e.g. datashader and the holoviews ecosystem), but you only mention matplotlib as a plotting backend.

erogluorhan replied:
We achieved this via Datashader and Holoviews actually. Our original plan was to put a separate section to this manuscript about that visualization work actually. Let us see if we can put together such a section during this review cycle. Would that be acceptable for you?

In the meantime, to clarify: The one significant work was for visualizing unstructured MPAS data. with Matplotlib, we were required to run much costly Delaunay triangulation on a set of about 43M data points that ended up  taking forever without an output. We made use of the connectivity (face-to-node connectivity) information that comes from the MPAS dataset and constructed Datashader triangular meshes and got them rendered. It was very much efficient (e.g. took a few minutes on a decent laptop, even without any Dask parallelization).

We put some of this details into the manuscript now to address your questions here, but it'd be great oto have a specific section and maybe visualization about this work if time permits.

TomNicholas replied:
That's interesting, thank you. I think it would be nice to devote a section to this if you have time, but the paper is also fine without a dedicated section.

TomNicholas commented on papers/orhan_eroglu/00_geocat.rst:
```diff
@@ -0,0 +1,446 @@
+
+- Fourier transforms such as band-block, band-pass, low-pass, and high-pass
+
+- Meteorological variables such as relative humidity, dew-point temperature,
+  heat index, saturation vapor pressure, and more
```
This functionality seems like it would have significant crossover with [MetPy](https://github.com/Unidata/MetPy). Is there scope for joining forces there?

erogluorhan replied:
Thanks for bringing this to attention! Yes, we are aware of this, and we have even joined forces to some extent so far: e.g. we migrated our own skewt parameters calculation to MetPy as we thought it was a better address for that, and we used a few functions from MetPy such as one dimensional interpolation to solve a more problem-specific case (hybrid to pressure level interpolation) etc. 

Further collaboration would definitely be possible and useful for the community though.

TomNicholas commented on papers/orhan_eroglu/00_geocat.rst:
```diff
@@ -0,0 +1,446 @@
+Science (SIParCS). Such programs are quite beneficial for both students and
+scientific software development teams. To exemplify, GeoCAT-examples and
+GeoCAT-viz in particular has received significant contributions through
+NCAR's Summer SIParCS in 2020 and 2021 summers (i.e. tens of visualization
+examples as well as improtant infrastructural changes were made available
```
This is great - do you have links to the output of those summer projects?

erogluorhan replied:
Thanks for asking! 

We have now added the references to the SIParCS efforts' publications at AMS and AGU. Other than that, the actual contributions of our past SIParCS interns have been deployed in our tools such as GeoCAT-comp and GeoCAT-examples. Since they are many, we can't think of an easy way to providing links to them; however, checking the GitHub records and stats to our repositories for dates between May and August for years 2020, 2021, and 2022 can show what contributions were made by our interns. We can also find a few NCAR presentations our interns gave at the end of their internship. We didn't think it'd be good to put them into the manuscript, but please let us know if you' d like to  see them (We can provide in this conversation).

TomNicholas reviewed (commented):
Thanks to the authors for this contribution, I'm sure GeoCAT will be useful to many users on a regular basis.

I've made a few specific comments, but overall I think the manuscript is clear, representative, and detailed.

deniederhut commented:
Thanks Tom!

deniederhut commented:
@scoobies mark pending comment

erogluorhan commented:
Hi @RBhupi and @TomNicholas , thank you very much for your great review! We are currently reading through your comments and will soon provide our revisions and responses here.

cbcunc commented on papers/orhan_eroglu/references.bib:
```diff
@@ -0,0 +1,169 @@
+@misc{pivot19,
```
Please include a DOI for each citation wherever humanly possible.

erogluorhan replied:
Thanks very much for reminding this! We have added several DOIs now. Please see the [6abdf11](https://github.com/scipy-conference/scipy_proceedings/pull/686/commits/6abdf113d11bc168b0369bf71e55ca5674b4b1f5) commit to see those added.

cbcunc replied:
We still need either a DOI, an ISSN, or a URL for all citations. Final edits are due July 4. A lot of these simply cannot be located.

erogluorhan replied:
Hi, there are a few types of citations in our bibliography. 

- For all of the online sources, we are providing URLs, and for some fo them, we are also providing DOIs.  

- For proceedings and journal papers, we are using however their bibtex entries come from Google Scholar. For many of these types of publications, neither Google Scholar nor the publishers themselves seem to be providing DOI or ISSN. For example:

 ```
@inproceedings{clyne2021project,
   title={Project Raijin: Community Geoscience Analysis Tools for Unstructured Grids},
   author={Clyne, John and Eroglu, Orhan and Medeiros, Brian and Zarzycki, Colin M},
   booktitle={AGU Fall Meeting 2021},
   year={2021},
   organization={AGU}
 }
```

[AGU](https://agu.confex.com/agu/fm21/meetingapp.cgi/Paper/901711) does not provide any ISSN or DOI for this proceedings publication as far as we know. 

Therefore, we are struggling to satisfy "either a DOI, an ISSN, or a URL for all citations".

Could you please clarify, or maybe exemplify this from a specific citation from our manuscript?

erogluorhan committed:
Avoid using GeoCAT-comp before defining it

erogluorhan committed:
Address @RBhupi's request for quantitative details about pivot to Python

erogluorhan committed:
Provide more details on wrapping up Fortran

erogluorhan committed:
Correct the term 'Meteorological variables'

RBhupi reviewed (approved):
Nice! Now, these statements are more concrete.

erogluorhan committed:
Apply @RBhupi's suggestion to shorten a sentence

erogluorhan committed:
Add geocat-datafiles section

RBhupi commented on papers/orhan_eroglu/00_geocat.rst:
```diff
@@ -69,16 +69,15 @@ GeoCAT team is committed to open development, which helps the team prioritize
-:cite:`ams22craker`. Additionally, GeoCAT also utilizes Numba, an open source
-just-in-time (JIT) compiler :cite:`numba`, to translate Python and NumPy code into
-machine codes in order to get faster executions wherever possible.
+GeoCAT has seven Python tools for geoscientific computation and visualization,
```
GeoCAT Python tools for geoscientific computation and visualization are built on cornerstone Pangeo :cite:`pangeo18` packages such as Xarray :cite:`xarray17`, and Dask :cite:`dask15`, as well as Jupyter Notebooks.

cbcunc commented:
Reviewers, please give your final recommendations. The deadline for final review comments is tomorrow.

Authors, please respond to review comments with changes or comments which resolve the conversation. Final acceptance decisions are July 4.

erogluorhan committed:
Address @TomNicholas's Dask vs Xarray terminology correction

erogluorhan committed:
Define open development earlier as @TomNicholas suggested

erogluorhan committed:
Fix the grammar

erogluorhan committed:
Separate Pangeo sentence from others

erogluorhan committed:
Add more details about interactive MPAS visualization notebook

erogluorhan committed:
Add citations to the GeoCAT efforts made by SIParCS interns

erogluorhan committed:
Try adding DOI to misc bib item

erogluorhan committed:
Add DOIs for references

erogluorhan committed:
Add one more DOI

RBhupi commented:
Thank you Co-Chairs @mepa @cbcunc @deniederhut @stargaser for inviting me to review the paper.

Independent Review Report
=========================

Reviewer: Bhupendra A. Raut

Department/Center/Division: Northwestern-Argonne Institute of Science and Engineering.

Institution/University/Company: Northwestern University.

Field of interest/expertise: Tropical Meteorology, Radar, and Satellite Data Analysis.

Country: USA


Article reviewed:  The Geoscience Community Analysis Toolkit: An Open Development, Community Driven Toolkit in the Scientific Python Ecosystem


GENERAL EVALUATION
------------------
The paper presents the current status of the python GeoCAT toolkit, established to provide a python alternative to NCL users in the geoscience community. The paper is relevant to the SciPy topics and contains some new details relevant to the geoscience community. The presentation is clear, and the length of the paper and figure captions are as per Scipy guidelines. The title and the abstract are pertinent and readable.
The authors provided an overview of the geoCAT components, plotting ability with examples, and community engagement efforts. The authors have made relevant changes suggested by both reviewers. New figures and subsections are added. This paper can be the beginning of the collaboration between GeoCAT and the MetPy community. Therefore, I recommend the publication of the paper in the SciPy-2022 proceedings.

- Quality of the approach: meets
- Quality of the writing: meets
- Quality of the figures/tables: meets


SPECIFIC EVALUATION
-------------------

For the following questions, please respond with 'yes' or 'no'.  If you answer 'no',  please provide a brief, one- to two-sentence explanation.

 - Is the code made publicly available and does the article  sufficiently describe how to access it?   
   > 'yes'

- Does the article present the problem in an appropriate context?
  Specifically, does it:

  - explain why the problem is important, 
  >yes
  - describe in which situations it arises,
  >yes
  - outline relevant previous work, 
  >yes
  - provide background information for non-experts
  > yes

- Is the content of the paper accessible to a computational scientist
  with no specific knowledge in the given field? 
  >yes

- Does the paper describe a well-formulated scientific or technical
  achievement? 
  >Yes. The project is ongoing and requires community engagement to achieve some of the objectives.

- Are the technical and scientific decisions well-motivated and
  clearly explained?
  > yes

- Are the code examples (if any) sound, clear, and well-written?
  > No code examples in the paper.

- Is the paper factually correct?
  > yes

- Is the language and grammar of sufficient quality?
  > yes

- Are the conclusions justified?
  > yes

- Is prior work properly and fully cited?
  > yes

- Should any part of the article be shortened or expanded? Please explain.
  > No.
- In your view, is the paper fit for publication in the conference proceedings?
  > yes 
- Please suggest specific improvements and indicate whether you think the article needs a significant rewrite (rather than a minor revision).
  > No

erogluorhan committed:
Fix broken references likely due to DOI

erogluorhan committed:
Add text for LOC comparison

erogluorhan committed:
Slight improvement to a sentence

erogluorhan committed:
Add a new figure, core technologies. Adjust naming of other figures.

erogluorhan committed:
Change Fig1 width to full paper width

erogluorhan committed:
Change Fig1 scale

erogluorhan committed:
Change Fig1 scale once again

cbcunc commented:
@TomNicholas Your final review comments are due today. Do you mark this as resolved or not?

erogluorhan committed:
Improve Figure 4

erogluorhan committed:
Add one more citation for Raijin

TomNicholas commented:
Hi, I'm happy that all of my comments have been addressed.

erogluorhan committed:
Add a paragraph and figure about interactive visualization notebook

erogluorhan committed:
Fix references

erogluorhan committed:
Debug the new notebook paragraph

erogluorhan committed:
Add text back and add the forgotten figure

erogluorhan committed:
Adjust Fig 7 to two-columns

deniederhut commented:
@cbcunc GitHub shows that you are still requesting changes on the paper. Is it ready for publication?

cbcunc commented:
@scoobies mark ready

deniederhut commented on papers/orhan_eroglu/references.bib:
```diff
@@ -0,0 +1,238 @@
+  organization  = "EGU General Assembly Conference Abstracts",
+}
+
+@article{xarray17,
```
Is this missing a DOI?

erogluorhan replied:
Added the DOI; thanks!

deniederhut commented on papers/orhan_eroglu/references.bib:
```diff
@@ -0,0 +1,238 @@
+  number    = "1",
+}
+
+@inproceedings{dask15,
```
This isn't correct. You can find the full citation here https://conference.scipy.org/proceedings/scipy2015/matthew_rocklin.html

erogluorhan replied:
Fixed it now; thanks very much!

deniederhut commented on papers/orhan_eroglu/references.bib:
```diff
@@ -0,0 +1,238 @@
+  pages         = "1--6",
+}
+
+ @misc{matplotlib,
```
matplotlib has a preferred citation - https://matplotlib.org/stable/users/project/citing.html

erogluorhan replied:
Thanks! Fixed this. Actually, I had checked https://matplotlib.org/devdocs/index.html but couldn't find a link to citation information; turns out I should have directly Googled it instead.

deniederhut reviewed (changes requested):
Would you mind taking a second pass over your references section? Some of these look incomplete or incorrect. I'm pretty sure SciPy hasn't published 130 volumes yet 😉

deniederhut commented:
@scoobies mark pending comment

erogluorhan committed:
Add DOI to xarray17 reference as @deniederhut suggested

erogluorhan committed:
Fix the dask15 reference as @deniederhut suggested

erogluorhan committed:
Fix the matplotlib reference as @deniederhut suggested

erogluorhan committed:
Convert Cartopy reference to their own suggestion

erogluorhan committed:
Fixed the forgotten PyPI reference to be cited

erogluorhan committed:
Fix a few references

erogluorhan committed:
Convert Pooch reference to their own suggestion

erogluorhan committed:
Fix the citation to the Pooch reference

erogluorhan committed:
Fix the Pooch reference

erogluorhan committed:
Fix the Pooch reference

erogluorhan commented:
> Would you mind taking a second pass over your references section? Some of these look incomplete or incorrect. I'm pretty sure SciPy hasn't published 130 volumes yet 😉

Thanks very much! We have addressed all of the inline suggestions as well as we have passed over our references again and converted some of them to the way that their owners suggest to cite (e.g. Cartopy, Sphinx, Pooch).

So, we believe it should be good to go now. @deniederhut

deniederhut commented:
@erogluorhan thanks for updating those! It looks like we're still missing DOIs for the numba and dyamond papers though?

erogluorhan committed:
Add DOI to numba and dyamond references

erogluorhan commented:
> @erogluorhan thanks for updating those! It looks like we're still missing DOIs for the numba and dyamond papers though?

Added now, thanks!

deniederhut reviewed (approved):
Thanks for adding those!

deniederhut commented:
@scoobies mark ready

erogluorhan commented:
Thanks very much, @deniederhut et al., for your great reviews that helped us improve our paper throughout!
